manager/controllers/app/select_modules.go

```diff
@@ -50,9 +50,16 @@ func (p *PathBuilder) solve() (Solution, error) {
 // Then, transformations are added to the found paths, and clusters are matched to satisfy restrictions from admin config policies.
 // Optimization is done by the shortest path (the paths are sorted by the length). To be changed in future versions.
 func (p *PathBuilder) FindPaths() []Solution {
+	var protocol taxonomy.ConnectionType
+	// If the connection name is empty, the default protocol is s3.
+	if p.Asset.DataDetails.Details.Connection.Name == "" {
```

Comment:
Wouldn't the entire structure be nil?

Reply:
It seems to be allocated anyway: https://github.com/fybrik/fybrik/blob/master/manager/controllers/app/fybrikapplication_controller.go#L367